src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx

```diff
@@ -71,38 +70,41 @@ export const Status: React.FC<StatusProps> = ({ accountId }) => {
     selectMarketDataByAssetIdUserCurrency(state, assetId ?? ''),
   )
 
-  const account = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
-
-  const serializedTxIndex = useMemo(() => {
-    if (!(state?.txid && accountId && account)) return ''
-    return serializeTxIndex(accountId, state.txid, account)
-  }, [state?.txid, accountId, account])
-
-  const confirmedTransaction = useAppSelector(gs => selectTxById(gs, serializedTxIndex))
-
   useEffect(() => {
-    if (!accountId) return
+    if (!(contextDispatch && state?.txid?.length && accountId)) return
+    ;(async () => {
+      // Ensuring we wait for the outbound Tx to exist
+      // Note, the transaction we wait for here is a Thorchain transaction, *not* the inbound Tx
+      const thorchainTxStatus = await waitForThorchainUpdate({
+        txId: state.txid!,
+        skipOutbound: false,
+      }).promise
 
-    if (confirmedTransaction && confirmedTransaction.status !== 'Pending' && contextDispatch) {
-      ;(async () => {
-        // Ensuring we wait for the outbound Tx to exist
-        // Note, the transaction we wait for here is a Thorchain transaction, *not* the inbound Tx
-        const thorchainTxStatus = await waitForThorchainUpdate({
-          txId: confirmedTransaction.txid,
-          skipOutbound: false,
-        }).promise
-
-        if ([TxStatusType.Confirmed, TxStatusType.Failed].includes(thorchainTxStatus)) {
-          contextDispatch({
-            type: ThorchainSaversWithdrawActionType.SET_WITHDRAW,
-            payload: {
-              txStatus: thorchainTxStatus === TxStatusType.Confirmed ? 'success' : 'failed',
-            },
-          })
+      if ([TxStatusType.Confirmed, TxStatusType.Failed].includes(thorchainTxStatus)) {
+        if (thorchainTxStatus === TxStatusType.Confirmed) {
+          appDispatch(
+            getOpportunitiesUserData.initiate(
+              [
+                {
+                  accountId,
+                  defiProvider: DefiProvider.ThorchainSavers,
+                  defiType: DefiType.Staking,
+                },
+              ],
+              { forceRefetch: true },
+            ),
+          )
```

Comment:
We want to dispatch this *before* `waitForThorchainUpdate`, as this will void what we're trying to do re: race condition

Reply:
Disregard me, got mixed with the tangent re: polling out of sync across the board, this is absolutely fine